### ISSUE TITLE
two issues about formatting

### ISSUE THREAD

**Issue by jamesfancy** [danschultequb/qub-typescript-xml-vscode#9]
I copied a HTML segment as sample into a XML file. Its content is here

``` xml
<?xml version="1.0" encoding="utf-8"?>

<head>
    <meta charset="UTF-8" />
    <title>Basic DataGrid - jQuery EasyUI Demo</title>

<link rel="stylesheet"
      type="text/css"
      href="http://www.jeasyui.net/easyui/themes/default/easyui.css" />
<link rel="stylesheet"
      type="text/css"
      href="http://www.jeasyui.net/easyui/themes/icon.css"/>
<link rel="stylesheet"
      type="text/css"
      href="http://www.jeasyui.net/demo/demo.css"/>

<script type="text/javascript"
        src="http://www.jeasyui.net/easyui/jquery.min.js"></script>
<script type="text/javascript"
        src="http://www.jeasyui.net/easyui/jquery.easyui.min.js"></script>
</head>
```

My purpose is to use the sample XML file for testing align-attributes. Good, alignment is OK. But I found another 2 issue:

1. there should be indents before `<link ...>` and `<script ...>`

2. `<script ....></script>` had been rewritten to `<script ... />` by formatter.  (Maybe it's not a issue for XML)

**Comment by danschultequb** [comment]
You're right about these issues.
1. I found a simplified version of your example:
```xml
<a>
<b/>
<c>hello</c>
<d/>
</a>
```
formats to:
```xml
<a>
    <b/>
<c>hello</c>
<d/>
</a>
```
which is definitely not correct. I'll get to fixing that right away.
2. Converting an element with no inner content to an empty element is definitely a good change in my opinion, but I respect that it's a good idea to have a setting to turn that off. I'll add a setting for it.

**Comment by danschultequb** [comment]
I just added the setting to turn off collapsing elements with no content in version 1.2.7.
I'm still looking into the formatting bug though.

**Comment by danschultequb** [comment]
I just fixed the formatting issue in version 1.2.8.

**Comment by jamesfancy** [comment]
All of the issues are OK. Thank you, @danschultequb!
